.github/workflows/try-runtime.yml

```diff
@@ -0,0 +1,53 @@
+name: try-runtime
+
+on:
+  pull_request:
+    types: [opened, synchronize, reopened]
+  workflow_dispatch:
+
+concurrency:
+  group: ${{ github.ref }}
+  cancel-in-progress: ${{ github.ref != 'refs/heads/master' }}
+
+jobs:
+  test-try-runtime:
+    strategy:
+      matrix:
+        runtime: [peregrine, spiritnet]
+    runs-on: ubuntu-latest
+    container:
+      image: paritytech/ci-unified:bullseye-1.77.0
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Set up caching for Cargo
+        uses: actions/cache@v4
+        with:
+          path: |
+            $HOME/.cargo/bin
+            $HOME/.cargo/registry/index/
+            $HOME/.cargo/registry/cache/
+            $HOME/.cargo/git/db/
+            target
+          key: ${{ runner.os }}-cargo-${{ hashFiles('**/Cargo.lock') }}
+          restore-keys: |
+            ${{ runner.os }}-cargo-
+
+      - name: script
+        run: |
+          echo "Running ${{ matrix.runtime }} runtime migration check"
+
+          echo "---------- Downloading try-runtime CLI ----------"
+          curl -sL https://github.com/paritytech/try-runtime-cli/releases/download/v0.7.0/try-runtime-x86_64-unknown-linux-musl -o try-runtime
+          chmod +x ./try-runtime
+          echo "Using try-runtime-cli version:"
+          ./try-runtime --version
+
+          echo "---------- Building ${{ matrix.runtime }} runtime ----------"
+          cargo build --release --locked -p ${{ matrix.runtime }}-runtime --features try-runtime
```

Comment:
Will each runtime get its own cache space?

Reply:
yes since I refere `${{ matrix.runtime }}` in cache key